src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1528,12 +1528,16 @@ def vmss_set(resource_group_name, name, no_wait=False, **kwargs):
         resource_group_name, name, raw=no_wait, **kwargs)
 
 
-def convert_av_set_to_managed_disk(resource_group_name, availability_set_name):
+def convert_av_set_to_managed_disk(resource_group_name, availability_set_name, platform_fault_domain_count=None):
     av_set = availset_get(resource_group_name, availability_set_name)
     if av_set.sku.name != 'Aligned':
         av_set.sku.name = 'Aligned'
+        if platform_fault_domain_count:
+            av_set.platform_fault_domain_count = platform_fault_domain_count
```

Comment:
Now that we have that API available, should we not retrieve the domain count and use the max supported value here?